### PR TITLE
docs: Update website to include SIGHUP, SIGTERM, and SIGINT behaviour

### PR DESCRIPTION
**Pull request by irenarindos**

### CODE REVIEW

File: website/content/docs/configuration/index.mdx [hunk 1]
```diff
@@ -87,6 +87,13 @@ path to the file.
 
 - [`plugins`](/boundary/docs/configuration/plugins): Configures options for plugins.
 
+## Signals
+On `SIGHUP`, worker and controller processes will reload their configuration file to pick up updated values
```
Comment (jefferai) [inline website/content/docs/configuration/index.mdx:91]: I'd add a caveat that not all configuration file values are able to be updated at runtime. Bonus points if you figure out which ones and ensure they're documented as such on the appropriate config pages :-)

Comment (irenarindos) [reply website/content/docs/configuration/index.mdx:91]: Thanks! I'll hunt these down

File: website/content/docs/configuration/controller.mdx [hunk 2]
```diff
@@ -106,6 +106,12 @@ description will be read.
   are anything specified by Go's [ParseDuration()](https://golang.org/pkg/time/#ParseDuration) method. Only
   used when an `ops` listener is set and the Controller is present. Default is 0 seconds.
 
+## Signals
+On `SIGHUP`, a controller will reload its configuration file to pick up any updates to its `database url`. All other updated values will be ignored.
```
Comment (Dan-Heath) [inline website/content/docs/configuration/controller.mdx:110]:
```suggestion
The `SIGHUP` signal causes a controller to reload its configuration file to pick up any updates to the `database url` value. Any other updated values are ignored.
```

File: website/content/docs/configuration/controller.mdx [hunk 3]
```diff
@@ -106,6 +106,12 @@ description will be read.
...
+
+`SIGTERM` or `SIGINT` will initiate graceful shutdown on a controller. Controller graceful shutdown
+will close listeners and servers before shutting down.
```
Comment (Dan-Heath) [inline website/content/docs/configuration/controller.mdx:113]:
```suggestion
The `SIGTERM` and `SIGINT` signals initiate a graceful shutdown on a controller. A graceful shutdown closes listeners and servers before shutting down the controller.
```

File: website/content/docs/configuration/index.mdx [hunk 4]
```diff
@@ -87,6 +87,17 @@ path to the file.
...
```
Comment (Dan-Heath) [inline website/content/docs/configuration/index.mdx:91]:
```suggestion
The `SIGHUP` signal causes worker and controller processes to reload their configuration files to pick up updated values.
```

File: website/content/docs/configuration/index.mdx [hunk 5]
```diff
@@ -87,6 +87,17 @@ path to the file.
...
+
+~> **Note:** Not all configuration values can be reloaded. Please see the configuration pages for
+[workers](/boundary/docs/configuration/worker/) and [controllers](/boundary/docs/configuration/controller) for details
+on which values can be reloaded on `SIGHUP`.
```
Comment (Dan-Heath) [inline website/content/docs/configuration/index.mdx:95]:
```suggestion
~> **Note:** You cannot reload all configuration values using the `SIGHUP` signal. Refer to the configuration pages for
[workers](/boundary/docs/configuration/worker/) and [controllers](/boundary/docs/configuration/controller) for details
on which values can be reloaded on `SIGHUP`.
```

File: website/content/docs/configuration/index.mdx [hunk 6]
```diff
@@ -87,6 +87,17 @@ path to the file.
...
+
+On `SIGTERM` or `SIGINT`, worker and controller processes will enter graceful shutdown. Controller graceful shutdown
+will close listeners and servers before shutting down. Worker graceful shutdown will wait for sessions to drain
+before shutting down. Workers in a graceful shutdown state will not receive new work, including session proxying, from the control plane.
```
Comment (Dan-Heath) [inline website/content/docs/configuration/index.mdx:99]:
```suggestion
The `SIGTERM` and `SIGINT` signals cause worker and controller processes to enter graceful shutdown. A graceful shutdown for a controller closes listeners and servers before shutting down the controller. A graceful shutdown for a worker waits for any sessions to drain
before shutting down the worker. Workers in a graceful shutdown state do not receive any new work, including session proxying, from the control plane.
```

File: website/content/docs/configuration/worker/index.mdx [hunk 7]
```diff
@@ -85,6 +85,13 @@ worker {
   `initial_upstreams`. This is currently only valid for workers using the PKI
   registration method and for workers directly connected to HCP Boundary.
 
+## Signals
+On `SIGHUP`, a worker will reload its configuration file to pick up updated values for `initial_upstreams` and `tags`.
+All other updated values will be ignored.
```
Comment (Dan-Heath) [inline website/content/docs/configuration/worker/index.mdx:90]:
```suggestion
The `SIGHUP` signal causes a worker to reload its configuration file to pick up any updates for the `initial_upstreams` and `tags` values.
Any other updated values are ignored.
```

File: website/content/docs/configuration/worker/index.mdx [hunk 8]
```diff
@@ -85,6 +85,13 @@ worker {
...
+
+`SIGTERM` or `SIGINT` will initiate graceful shutdown on a worker. The worker will wait for sessions to drain
+before shutting down. Workers in a graceful shutdown state will not receive new work, including session proxying, from the control plane.
```
Comment (Dan-Heath) [inline website/content/docs/configuration/worker/index.mdx:93]:
```suggestion
The `SIGTERM` and `SIGINT` signals initiate a graceful shutdown on a worker. The worker waits for any sessions to drain
before shutting down. Workers in a graceful shutdown state do not receive any new work, including session proxying, from the control plane.
```

**Review by Dan-Heath — COMMENTED** [review]
I added some suggestions for minor rewrites, but overall the content makes sense and sounds good to me. Please let me know if you have any questions or if there's anything that I can do to help. Thanks very much for this update!

File: website/content/docs/configuration/index.mdx (hunk 5 above)
Comment (irenarindos) [reply website/content/docs/configuration/index.mdx:95]: thanks for all these!!

**Review by Dan-Heath — APPROVED** [review]
LGTM, thanks!!

**Pull request merged**